### ISSUE TITLE
server,rpc: in v22.2, the RPC heartbeat server incorrectly performs version check on CLI clients

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#90131]
We have a cross-version compatibility guarantee between CLI clients and servers, such that it should be possible to use version N-1 of the CLI to access version N servers, and vice-versa.

There is a regression for this in v22.2: it is not any more possible to use a v22.1 CLI to operate a v22.2 server.

For example, `cockroach node drain` fails with:
```
ERROR: failed to connect to the node: initial connection heartbeat failed: rpc error: code = Unknown desc = version compatibility check failed on ping request: cluster requires at least version 22.1-74, but peer has version 22.1
```

**Comment by knz** [comment]
Investigated this further - this seems to happen in 22.1 too so not a GA blocker.

**Comment by nvanbenschoten** [comment]
@knz do you know what change introduced the regression?

**Comment by knz** [comment]
Yeah it was the big init refactor we did back in 21.1 (or maybe 20.2, I'm not sure), the one that allowed us to simplify/strengthen the upgrade process.
